refactron/llm/orchestrator.py

```
@@ -162,6 +258,117 @@ def generate_suggestion(self, issue: CodeIssue, original_code: str) -> Refactori
+        if not issues:
+            raise ValueError("No issues provided for batch suggestion")
+
+        # 1. Retrieve Context
+        context_snippets = []
+        if self.retriever:
+            try:
+                # Use the first few issues for context retrieval
+                query = " ".join([i.message for i in issues[:3]])
+                results = self.retriever.retrieve_similar(query, top_k=3)
+                context_snippets = [r.content for r in results]
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)